F-FrontEnd/src/F-compile-decl.c

```diff
@@ -2309,11 +2309,15 @@ declare_id_type(ID id, TYPE_DESC tp)
             tpp = tp;
             while(TYPE_REF(tpp) != NULL && IS_ARRAY_TYPE(TYPE_REF(tpp)))
                 tpp = TYPE_REF(tpp);
-            if (TYPE_REF(tpp) != NULL && !type_is_soft_compatible(tq, TYPE_REF(tpp)))
-                goto no_compatible;
-            if (TYPE_REF(tpp) == NULL ||
-                type_is_specific_than(tq, TYPE_REF(tpp)))
-                TYPE_REF(tpp) = tq;
+            if (TYPE_REF(tpp) != NULL 
+                && !type_is_soft_compatible(tq, TYPE_REF(tpp))) 
+            {
+                if(TYPE_BASIC_TYPE(TYPE_REF(tpp)) == TYPE_FUNCTION) {
```

Comment:
@clementval Still I'm wondering if the following two conditions are equivalent or the second one is more precise.

```
type_is_specific_than(tq, TYPE_REF(tpp))
```
```
TYPE_BASIC_TYPE(TYPE_REF(tpp)) == TYPE_FUNCTION
```

Is there no need to make the condition in line 2315 as follows?

```
type_is_specific_than(tq, TYPE_REF(tpp) || TYPE_BASIC_TYPE(TYPE_REF(tpp)) == TYPE_FUNCTION
```

Reply:
@h-murai I think you are right. I updated the condition and the tests are still passing.